### PR TITLE
[22.3] Fix `javax.crypto.JceSecurity` substitutions in JDK >= 17.0.10

### LINKED ISSUE #608: [22.3] Mandrel 22.3 fails to build with JDK 17.0.10-EA
## Description
Mandrel 22.3 fails to build with JDK 17.0.10-EA with error:

```
Error: Substitution target for com.oracle.svm.core.jdk.Target_javax_crypto_JceSecurity_IdentityWrapper is invalid as inner class IdentityWrapper in javax.crypto.JceSecurity can not be found. Make sure that the inner class is present.
```

This is due to the upstream JDK 17 change in https://github.com/openjdk/jdk17u/commit/a06047acce82f60b5ca193a7b2aa329ed24b46f4

### PR DESCRIPTION
**Pull request by zakkak**
Closes https://github.com/graalvm/mandrel/issues/608

### CODE REVIEW

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java [hunk 1]
```diff
@@ -317,6 +319,9 @@ static boolean isTrustedCryptoProvider(Provider provider) {
 @SuppressWarnings({"unused"})
 final class Target_javax_crypto_JceSecurity {
 
+    @Alias @TargetElement(onlyWith = JDK17_0_10OrLater.class)//
+    public static ReferenceQueue<Object> queue;
```
Comment (jerboaa) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java]: Same comment as for the 23.0 PR with respect to the update version conditionals.

**Comment by jerboaa** [comment]
CI failures:

- smallrye graphql: https://github.com/quarkusio/quarkus/issues/36975
- Other windows issues are related to extended socket options new in 17.0.10: #612

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java (hunk 1 above)
Comment (zakkak) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java]: Done

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java [hunk 2]
```diff
@@ -848,9 +850,31 @@ private Function<Object, Object> constructVerificationCacheCleaner(Class<?> jceS
             };
         }
         /*
-         * For JDK 17 and later, the verification cache is an IdentityWrapper -> Verification result
-         * ConcurrentHashMap. The IdentityWrapper contains the actual provider in the 'obj' field.
+         * For JDK 17.0.10 and later, the verification cache is a WeakIdentityWrapper ->
+         * Verification result ConcurrentHashMap. The WeakIdentityWrapper contains the actual
+         * provider in the 'obj' field.
          */
+        if (JavaVersionUtil.JAVA_SPEC == 17 && GraalServices.getJavaUpdateVersion() >= 10) {
```
Comment (jerboaa) [inline substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java]: `new QueueFieldPresent().getAsBoolean()` instead?

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java [hunk 3]
```diff
@@ -432,7 +482,12 @@ static Object providerKey(Provider p) {
         if (JavaVersionUtil.JAVA_SPEC <= 11) {
             return p;
         }
+
         /* Starting with JDK 17 the verification results map key is an identity wrapper object. */
+        if (JavaVersionUtil.JAVA_SPEC == 17 && GraalServices.getJavaUpdateVersion() >= 10) {
```
Comment (jerboaa) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java]:
```
if (new WeakIdentityWrapperPresent().getAsBoolean()) {
...
}
```

File: substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java (hunk 2 above)
Comment (zakkak) [reply substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/SecurityServicesFeature.java]: Done

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java (hunk 3 above)
Comment (zakkak) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SecuritySubstitutions.java]: Done

**Review by jerboaa — APPROVED** [review]

**Pull request merged**